### ISSUE TITLE
1.5/Win8: Statistics

### ISSUE THREAD

**Issue by AnthorNet** [Marginal/EDMarketConnector#10]
Can't read them, as the window is too small.
![sans-titre-2](https://cloud.githubusercontent.com/assets/654353/8306080/51928c66-19b5-11e5-9743-317283c8904b.jpg)

**Comment by Marginal** [comment]
Do you have Python 2.7 (plus requests) installed? If so, please run from the command-line and see if there are any error messages. Also, if you don't mind sharing your cmdr details, please send me the .json file that it dumps.

**Comment by AnthorNet** [comment]
```
Traceback (most recent call last):
  File "D:\EDMarketConnector-rel-150\stats.py", line 159, in showstats ('Average profit', data['stats']['blackMarket']['profit'] / data['stats']['blackMarket']['count']),
ZeroDivisionError: integer division or modulo by zero
Traceback (most recent call last):
  File "D:\EDMarketConnector-rel-150\stats.py", line 212, in showstats ('Commodities stolen',       str(data['stats']['crime']['stolenCargo']['qty'])),
KeyError: 'stolenCargo'
Traceback (most recent call last):
  File "D:\EDMarketConnector-rel-150\stats.py", line 223, in showstats for key in sorted(data['ships'].keys(), key=int):
AttributeError: 'list' object has no attribute 'keys'
```

**Comment by Marginal** [comment]
Thanks. Looks like I need to code this much more defensively.

**Comment by AnthorNet** [comment]
;)

**Comment by Marginal** [comment]
Should be fixed in [Release 1.51](https://github.com/Marginal/EDMarketConnector/releases/tag/rel-151) - please let me know.
Also, please check if your ship(s) are listed on the Ships tab - your data structure appears to be different to mine.

**Comment by wolverine2710** [comment]
I basically had the exact had the exact same screen as Anthornet with 1.50. With 1.51 I see a different screen , bigger but still to small.

Python version:
$ "C:\Data\Programming languages\64 bits\Python27\python.exe" --version
Python 2.7.10

![edmc_151_main_window](https://cloud.githubusercontent.com/assets/4365515/8316629/9b13edb0-19f8-11e5-8965-7ccc255e8bd3.jpg)
![edmc_151_fetching_data_screen](https://cloud.githubusercontent.com/assets/4365515/8316635/a052bd7e-19f8-11e5-8e14-038d2a17fb44.jpg)
![edmc_151_stats_screen](https://cloud.githubusercontent.com/assets/4365515/8316637/a40bbda8-19f8-11e5-854f-86d0ab753b77.jpg)






​
My gitbash window shows the following:

$ "C:\Data\Programming languages\64 bits\Python27\python.exe" EDMarketConnector
.py
Loadout: 128064033: Unknown module "PowerPlant"
Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\Data\Programming languages\64 bits\Python27\lib\lib-tk\Tkinter.py", l
ine 1536, in __call__
    return self.func(*args)
  File "EDMarketConnector.py", line 93, in <lambda>
    file_menu.add_command(label="Statistics", command=lambda:stats.StatsDialog(s
elf.w, self.session))
  File "c:\Data\Games\Elite Dangerous\edmc\stats.py", line 59, in __init__
    self.showstats()
  File "c:\Data\Games\Elite Dangerous\edmc\stats.py", line 91, in showstats
    StatsResults(self.parent, data)
  File "c:\Data\Games\Elite Dangerous\edmc\stats.py", line 170, in __init__
    self.addstat(page, data, *thing)
  File "c:\Data\Games\Elite Dangerous\edmc\stats.py", line 349, in addstat
    assert False, content
AssertionError: (['stats', 'trade', 'profit'], ['stats', 'trade', 'count'])

The fetching data window never disappears and  clicking update in the main window does not nothing. Seems to hang.

Orphu of EDAPI also shows some stats. It worked fine for him, all kind of errors for me. It turned out that his and mine JSON was a bit different. Perhaps because I had cleared my save and I had a PP JSON. His save was not cleared, it was from before PP but when he ran  EDAPI he was running PP. Before Orphu could help you with this. Not sure what stats you are going to display but biobob has a private tools which basically display everything which is described in the section "What data is in the JSON datastructure" of this Redit thread . Perhaps you are displaying the same info ;-)

**Comment by wolverine2710** [comment]
The link got lost. Again, hopefully now in correct markdown markup.

"What data is in the JSON datastructure" of [this Redit thread](http://www.reddit.com/r/EliteDangerous/comments/38ireb/mobile_api_based_tools_ideal_for_trading_logbook/).

**Comment by Marginal** [comment]
Getting there. Have updated the [draft release](https://github.com/Marginal/EDMarketConnector/releases/tag/rel-151).

**Comment by AnthorNet** [comment]
Works fine for me now ^^

**Comment by Marginal** [comment]
Thanks.
Fixed in [Release 1.51](https://github.com/Marginal/EDMarketConnector/releases/tag/rel-151) (RC2)
